### PR TITLE
Update code in TimestampBehavior

### PR DESCRIPTION
**Pull request by mehdi-fathi**
Update code in TimestampBehavior for increase readbility.

### CODE REVIEW

File: src/ORM/Behavior/TimestampBehavior.php [hunk 1]
```diff
@@ -136,8 +136,9 @@ public function implementedEvents()
      */
     public function timestamp(DateTime $ts = null, $refreshTimestamp = false)
     {
-        if ($ts) {
-            if ($this->_config['refreshTimestamp']) {
+        if ($ts instanceof DateTime) {
```
Comment (dereuromark) [inline src/ORM/Behavior/TimestampBehavior.php]: Should this be DateTimeInterface?

Comment (ndm2) [inline src/ORM/Behavior/TimestampBehavior.php]: I'd say if changing this all, then it should be an explicit not-`null` comparison, as the argument typhint restricts things to either `DateTime` or `null`, ie no other value can get there in the first place.

**Comment by dereuromark** [comment]
What does this PR exactly do/change?
The existing check looked OK to me (it can either be null or an object, right?).

**Comment by ADmad** [comment]
Closing as none of the changes done improve or fix anything.

**Comment by mehdi-fathi** [comment]
we have 
`if ($class instanceof SessionHandlerInterface) {
            return $this->_engine = $class;
        }`
in 

> Cake\Network\Session.php line 245

**Comment by ndm2** [comment]
@mehdi-fathi That is because unlike the `$ts` argument of the `TimestampBehavior::timestamp()` method, the `$class` argument of `Session::engine()` is not type-hinted, as it is also allowed to be a string.

Given that PHP does not allow union type-hints and doesn't support argument based method overloading, this is how things are beig handled for arguments that accept different types. You may want to check [**the PHP manual on Type declarations**](http://php.net/manual/en/functions.arguments.php#functions.arguments.type-declaration) for more information on how type-hinting in PHP works.

**Pull request closed without merging**